File: effective/src/main/kotlin/com/effective/chapter/_2/hierarchicalbuilder/java/PizzaTest.java
package com.effective.chapter._2.hierarchicalbuilder.java;


import static _2.hierarchicalbuilder.java.NyPizza.Size.SMALL;
import static _2.hierarchicalbuilder.java.Pizza.Topping.HAM;
import static _2.hierarchicalbuilder.java.Pizza.Topping.ONION;
import static _2.hierarchicalbuilder.java.Pizza.Topping.SAUSAGE;


// 계층적 빌더 사용 (21쪽)
public class PizzaTest {

  public static void main(String[] args) {
    NyPizza pizza = new NyPizza.Builder(SMALL)
        .addTopping(SAUSAGE)
        .addTopping(ONION).build();

    Calzone calzone = new Calzone.Builder()
        .addTopping(HAM).sauceInside().build();

    System.out.println(pizza);
    System.out.println(calzone);
  }
}

File: effective/src/main/kotlin/com/effective/chapter/_2/hierarchicalbuilder/PizzaTest.kt
package com.effective.chapter._2.hierarchicalbuilder

object PizzaTest {
    @JvmStatic
    fun main(args: Array<String>) {
        val pizza = NyPizza(
            toppings = setOf(Topping.SAUSAGE, Topping.ONION),
            size = NyPizza.Size.SMALL
        )
        val calzone = Calzone(
            toppings = setOf(Topping.HAM),
            sauceInside = true
        )
        println(pizza)
        println(calzone)
    }
}